arch/risc-v/src/esp32c3/esp32c3_irq.c

```diff
@@ -166,6 +189,52 @@ void up_disable_irq(int cpuint)
   leave_critical_section(irqstate);
 }
 
+/****************************************************************************
+ * Name: esp32c3_bind_irq
+ *
+ * Description:
+ *   Bind IRQ and resource with given parameters.
+ *
+ * Input Parameters:
+ *   cpuint    - CPU interrupt ID
+ *   periphid  - Peripheral ID
+ *   prio      - Interrupt priority
+ *   flags     - Interrupt flags
+ *
+ * Returned Value:
+ *   Allocated CPU interrupt on success, a negated error on failure.
+ *
+ ****************************************************************************/
+
+int esp32c3_bind_irq(uint8_t cpuint, uint8_t periphid, uint8_t prio,
```

Comment:
Why was this function created? Isn't doing the same thing as `esp32_request_irq`?

Reply:
The reason is WiFi use specific CPU interrupt ID, and WiFi will bind multi WiFi peripheral interrupt to this CPU interrupt. Maybe BT/BLE has same action.

Reply:
I add reserved CPU interrupt for WiFi: https://github.com/apache/incubator-nuttx/pull/3040/files#diff-0c00cebfe74b712116ea0e572904991de5f9c97c89273e53e282fa799506d048R104, later will also reserve for BT.

Reply:
My issue is this function is almost exactly the same as `esp32c3_irq_request`.  It just bypasses the interrupt to peripheral map.

Reply:
Yes, I will modify the `esp32c3_irq_request`, and let it call `esp32c3_bind_irq` with allocated CPU int ID.